source/fundamentals/poco.txt

```diff
@@ -0,0 +1,249 @@
+can direct the driver to store the ``YearBuilt`` property of the
+``House`` object as ``year_built`` in the serialized BSON document:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       public ObjectId Id { get; set; }
+
+       [BsonElement("year_built")]
+       public int YearBuilt { get; set; }
+   }
+
+Select Type Representation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For some {+language+} primitive types you can control the BSON type used
+to represent the value of your POCO property using the
+``BsonRepresentationAttribute``. The following code shows how you can
+specify that the ``YearBuilt`` property, defined as a ``char`` in
+{+language+}, is serialized as a BSON ``Int32`` type:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       public ObjectId Id { get; set; }
+
+       [BsonRepresentation(BsonType.Int32)]
+       public char YearBuilt { get; set; }
+   }
+
+Set Field Order
+~~~~~~~~~~~~~~~
+
+The driver serializes properties to BSON fields in the order they
+are specified in the POCO. To store properties in a custom order, you
+can specify the ``Order`` named parameter in the ``BsonElement``
+attribute. The following code shows how you can direct the driver to
+store the ``YearBuilt`` property after the ``Style`` property:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       public ObjectId Id { get; set; }
+
+       [BsonElement(Order = 2)]
+       public int YearBuilt { get; set; }
+
+       [BsonElement(Order = 1)]
+       public string Style { get; set; }
+   }
+
+Any properties that do not have an explicit ``Order`` will occur after
+those that do.
+
+Identify Id Property
+~~~~~~~~~~~~~~~~~~~~
+
+By default, the driver uses any public property named ``Id``, ``id``, or ``_id`` as the
+identifier to map to the BSON ``_id`` field. You can explicitly select the
+identifier using the ``BsonIdAttribute``. The following code shows how
+you can direct the driver to map the ``Identifier`` property to the ``_id`` field:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       [BsonId]
+       public string Identifier { get; set; }
+   }
+
+Omit Empty Fields
+~~~~~~~~~~~~~~~~~
+
+By default, the driver serializes undefined properties to fields with ``null``
+values. You can direct the driver to ignore empty properties using the
+``BsonIgnoreAttribute``. The following code shows how you can prevent
+the driver from serializing the ``YearBuilt`` property if it is empty:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       public ObjectId Id { get; set; }
+
+       [BsonIgnore]
+       public int YearBuilt { get; set; }
+       public string Style { get; set; }
+   }
+
+Example
+-------
+
+The following example shows how the {+driver-short+} inserts a document
+instantiated as a POCO with the following custom field mapping specifications:
```

Comment:
Issue: This feels a little long and confusing, but I also struggled with cleaning it up! 

```suggestion
The following example shows how to define a C# class with custom field mapping specifications, create a POCO to represent a document, then insert the POCO into MongoDB. 
```